lib/helpers/getPrerenderManifest.js

```diff
@@ -1,9 +1,46 @@
 const { join } = require("path");
 const { readJSONSync } = require("fs-extra");
 const { NEXT_DIST_DIR } = require("../config");
+const nextConfig = require("./getNextConfig")();
+const getDataRouteForRoute = require("./getDataRouteForRoute");
+
+const transformManifestForI18n = (manifest) => {
+  const { routes, dynamicRoutes } = manifest;
+  const { defaultLocale, locales } = nextConfig.i18n;
+  const newRoutes = {};
+  Object.entries(routes).forEach(
+    ([route, { dataRoute, initialRevalidateSeconds, srcRoute }]) => {
+      const isDynamicRoute = !!srcRoute;
+      if (isDynamicRoute) {
+        newRoutes[route] = routes[route];
+      } else {
+        locales.forEach((locale) => {
+          const routeWithPrependedLocale = `/${locale}${route}`;
+          newRoutes[routeWithPrependedLocale] = {
+            dataRoute: getDataRouteForRoute(route, locale),
+            srcRoute: route,
+            initialRevalidateSeconds: false,
+          };
+        });
+      }
+    }
+  );
+  const newDynamicRoutes = {};
+  Object.entries(dynamicRoutes).forEach(([route, { dataRoute, fallback }]) => {
+    newDynamicRoutes[route] = {
+      route,
+      dataRoute: getDataRouteForRoute(route, defaultLocale),
```

Comment:
What is the reason for overwriting dataRoute for all entries in dynamicRoutes?

Reply:
they're not "right" as far as i can tell. the cypress tests fail for dynamicRoutes when we leave it as the dataRoute generated in the manifest (vs using the dataRoute of the defaultLocale)

Reply:
update: it's because i wasn't handling locale fallback cases separately, removing